.github/workflows/go.yml

```diff
@@ -6,15 +6,20 @@ on:
   pull_request:
     branches: [ main ]
 
+permissions:
+  contents: read
```

Comment:
Curious as to why this was added? I've had this flagged by zimzor in other repos for it being too broad.

Reply:
Oh, I think I know why 😬

Reply:
This permissions block _reduces_ the scope the workflow has access to, is that correct?

Reply:
When running `zizmor` without this, I get the following:

```
warning[excessive-permissions]: overly broad permissions
  --> ./.github/workflows/go.yml:11:3
   |
11 | /   build:
12 | |     runs-on: ubuntu-latest
...  |
29 | |     - name: Benchmark
30 | |       run: make benchmark
   | |                          -
   | |__________________________|
   |                            this job
   |                            default permissions used due to no permissions: block
   |
   = note: audit confidence → Medium
```

Considering it doesn't affect the outcome of running the benchmarks, it seemed safe to include.

Reply:
Perfect, thanks! I'll add this to some of the other workflows I'm working through